src/pages/components/file-upload.mdx

```diff
@@ -0,0 +1,96 @@
+---
+title: File upload
+description: >
+  File upload lets users select and upload files.
+package: '@zendeskgarden/react-forms'
+components:
+  - forms/src/elements/FileUpload.tsx
+---
+
+<Usage>
+  <Use>
+    <ul>
+      <li>To let users attach a file or image to something else</li>
+      <li>To let users add images to their avatars</li>
+    </ul>
+  </Use>
+</Usage>
+
+## How to use it
+
+### Default
+
+Users drag and drop files on to the target or click to select which files to upload.
+
+import Default from '../../examples/forms/file-upload/FileUpload.tsx';
+import DefaultCode from '!!raw-loader!../../examples/forms/file-upload/FileUpload.tsx';
+
+<CodeExample code={DefaultCode}>
+  <Default />
+</CodeExample>
+
+### Disabled
+
+A disabled File upload prevents user interaction. It doesn’t appear in the tab order and can’t receive focus.
+
+import Disabled from '../../examples/forms/file-upload/Disabled.tsx';
+import DisabledCode from '!!raw-loader!../../examples/forms/file-upload/Disabled.tsx';
+
+<CodeExample code={DisabledCode}>
+  <Disabled />
+</CodeExample>
+
+### Size
+
+File upload can be default or compact. Size affects spacing between the label/description and the touch target.
+
+import Size from '../../examples/forms/file-upload/Size.tsx';
+import SizeCode from '!!raw-loader!../../examples/forms/file-upload/Size.tsx';
+
+<CodeExample code={SizeCode}>
+  <Size />
+</CodeExample>
+
+## How to use it well
+
+### Put file requirements in hints
+
+<BestPractice>
+  <Do imageSource={props.data.fileUploadRequirements.childFile.childImageSharp.fluid} imageIsSquare>
+    <p>
+      <Markdown>Use `Hint` to describe image requirements and restrictions</Markdown>
+    </p>
+  </Do>
+</BestPractice>
```

Comment:
@steue I'm wondering if this section is ready for prime time? Given the size, it is nearly unreadable. Or do you think we'll have a very rapid follow-on fix specified?

Reply:
i don't think it's ready, and i'm not sure if there's a speedy fix on the way. these image examples have been a repeated thorn in the foot and i think it's worth revisiting how to better design/implement these.

Reply:
Removed for now.